iceoryx_posh/source/capro/service_description.cpp

```diff
@@ -104,86 +101,38 @@ ServiceDescription::ServiceDescription(const cxx::Serialization& f_serial) noexc
 }
 
 ServiceDescription::ServiceDescription() noexcept
-    : ServiceDescription(InvalidID, InvalidID, InvalidID)
+    : ServiceDescription(InvalidString, InvalidString, InvalidString)
 {
 }
 
-ServiceDescription::ServiceDescription(uint16_t f_serviceID, uint16_t f_instanceID) noexcept
-    : ServiceDescription(f_serviceID, InvalidID, f_instanceID)
-{
-    m_hasServiceOnlyDescription = true;
-}
-
-ServiceDescription::ServiceDescription(const IdString_t& f_service, const IdString_t& f_instance) noexcept
-    : ServiceDescription(f_service, f_instance, InvalidIDString)
-{
-    m_hasServiceOnlyDescription = true;
-}
-
-ServiceDescription::ServiceDescription(uint16_t f_serviceID, uint16_t f_eventID, uint16_t f_instanceID) noexcept
-    : m_serviceID(f_serviceID)
-    , m_eventID(f_eventID)
-    , m_instanceID(f_instanceID)
-    , m_serviceString(cxx::TruncateToCapacity, cxx::convert::toString(f_serviceID))
-    , m_instanceString(cxx::TruncateToCapacity, cxx::convert::toString(f_instanceID))
-    , m_eventString(cxx::TruncateToCapacity, cxx::convert::toString(f_eventID))
-{
-}
-
-ServiceDescription::ServiceDescription(const IdString_t& f_service,
-                                       const IdString_t& f_instance,
-                                       const IdString_t& f_event,
-                                       ClassHash f_classHash,
+ServiceDescription::ServiceDescription(const IdString_t& service,
+                                       const IdString_t& instance,
+                                       const IdString_t& event,
+                                       ClassHash classHash,
                                        Interfaces interfaceSource) noexcept
-    : m_serviceString{f_service}
-    , m_instanceString{f_instance}
-    , m_eventString{f_event}
-    , m_classHash(f_classHash)
+    : m_serviceString{service}
+    , m_instanceString{instance}
+    , m_eventString{event}
+    , m_classHash(classHash)
     , m_interfaceSource(interfaceSource)
 {
-    if (!(cxx::convert::stringIsNumber(m_serviceString.c_str(), cxx::convert::NumberType::UNSIGNED_INTEGER)
-          && cxx::convert::fromString(m_serviceString.c_str(), m_serviceID)))
-    {
-        m_serviceID = InvalidID;
-    }
-
-    if (!(cxx::convert::stringIsNumber(m_instanceString.c_str(), cxx::convert::NumberType::UNSIGNED_INTEGER)
-          && cxx::convert::fromString(m_instanceString.c_str(), m_instanceID)))
-    {
-        m_instanceID = InvalidID;
-    }
-
-    if (!(cxx::convert::stringIsNumber(m_eventString.c_str(), cxx::convert::NumberType::UNSIGNED_INTEGER)
-          && cxx::convert::fromString(m_eventString.c_str(), m_eventID)))
-    {
-        m_eventID = InvalidID;
-    }
 }
 
 bool ServiceDescription::operator==(const ServiceDescription& rhs) const
```

Comment:
We could also consider something like this in order to prevent RouDi to connect pub/subs with invalid `ServiceDescription`'s and omit some of the `isValid()` checks. However, I'm not convinced that this makes sense semantically, so I decided against.

```
bool ServiceDescription::operator==(const ServiceDescription& rhs) const
{
    if (!isValid() || !rhs.isValid())
    {
        return false;
    }
    // .. snip ...   
}
```

Reply:
After a discussion with @MatthiasKillat I'm tempted to implement the above code :grimacing:  @elBoberido What do you think?